bin/find-spark-home.cmd

```diff
@@ -0,0 +1,56 @@
+@echo off
+
+rem
+rem Licensed to the Apache Software Foundation (ASF) under one or more
+rem contributor license agreements.  See the NOTICE file distributed with
+rem this work for additional information regarding copyright ownership.
+rem The ASF licenses this file to You under the Apache License, Version 2.0
+rem (the "License"); you may not use this file except in compliance with
+rem the License.  You may obtain a copy of the License at
+rem
+rem    http://www.apache.org/licenses/LICENSE-2.0
+rem
+rem Unless required by applicable law or agreed to in writing, software
+rem distributed under the License is distributed on an "AS IS" BASIS,
+rem WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+rem See the License for the specific language governing permissions and
+rem limitations under the License.
+rem
+
+rem Path to Python script finding SPARK_HOME
+set FIND_SPARK_HOME_PYTHON_SCRIPT=%~dp0find_spark_home.py
+
+rem Default to standard python interpreter unless told otherwise
+set PYTHON_RUNNER=python
+rem If PYSPARK_DRIVER_PYTHON is set, it overwrites the python version
+if not "x%PYSPARK_DRIVER_PYTHON%"=="x" (
+  set PYTHON_RUNNER=%PYSPARK_DRIVER_PYTHON%
+)
+rem If PYSPARK_PYTHON is set, it overwrites the python version
+if not "x%PYSPARK_PYTHON%"=="x" (
+  set PYTHON_RUNNER=%PYSPARK_PYTHON%
+)
+
+rem If there is python installed, trying to use the root dir as SPARK_HOME
+where %PYTHON_RUNNER%
+if %ERRORLEVEL% neq 0 (
+  echo %PYTHON_RUNNER% wasn't found; Python doesn't seem to be installed
+    if "x%SPARK_HOME%"=="x" (
+      set SPARK_HOME=%~dp0..
+    )
+)
```

Comment:
I think the problem here from the last commit is:

- now `PYTHON_RUNNER` can't be an absolute path as `where` does not work with it

  ```
  ERROR: Invalid pattern is specified in "path:pattern".
  C:\Python27\python.exe wasn't found; Python doesn't seem to be installed
  ```

- It print out the output from `where`:

  ```
  C:\...>pyspark
  C:\cygwin\bin\python
  C:\Python27\python.exe
  ...
  ```

- and the error message looks not more useful than the previous one:

  ```
  python wasn't found; Python doesn't seem to be installed
  ```